CMakeLists.txt

```diff
@@ -0,0 +1,79 @@
+##===----------------------------------------------------------------------===##
+##
+## This source file is part of the Swift open source project
+##
+## Copyright (c) 2024 Apple Inc. and the Swift project authors
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.md for the list of Swift project authors
+##
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+
+cmake_minimum_required(VERSION 3.24)
```

Comment:
Is this for CI? It might be worth trying this build across a few versions of CMake and doing something like this
```cmake
cmake_minimum_required(VERSION 3.24...3.29)
```

That way we pick up CMP0156 and CMP0157. We definitely want these two if folks have a new enough CMake since it will fix the linker warning about duplicate libraries and will result in faster incremental builds due to the better Swift build graphs in CMake 3.29.

At the very least, between the `cmake_minimum_required` and `project` invocations, can you enable those policies? Something like this:

```cmake
cmake_minimum_required(VERSION 3.24)

if(POLICY CMP0156)
    # Deduplicate linked libraries where appropriate
    cmake_policy(SET CMP0156 NEW)
endif()

if(POLICY CMP0157)
    # New Swift build model: improved incremental build performance and LSP support
    cmake_policy(SET CMP0157 NEW)
endif()

project(SwiftFoundationICU
    LANGUAGES CXX Swift)
...
```